### ISSUE TITLE
Change README rule of thumb

### ISSUE THREAD

**Issue by ChrisRackauckas** [JuliaArrays/StaticArrays.jl#506]
The README says:

>Note that in the current implementation, working with large StaticArrays puts a lot of stress on the compiler, and becomes slower than Base.Array as the size increases. A very rough rule of thumb is that you should consider using a normal Array for arrays larger than 100 elements. For example, the performance crossover point for a matrix multiply microbenchmark seems to be about 11x11 in julia 0.5 with default optimizations.

In Julia v1.0 there were some pretty big changes to tuple optimizations, for example https://github.com/JuliaLang/julia/pull/27398 . Since SVectors are backed by tuples, this seemed to have a pretty big effect on large static arrays. For example, 11x11 used to be over the tuple inference limit, counter acted by some of the nice generated functions in StaticArrays.jl, and that made them seem to match. But without that inference limit, 11x11 seems to be much better than before.

@dpsanders mentioned in Slack

```
Chris Rackauckas [6:43 PM]

>For example, the performance crossover point for a matrix multiply microbenchmark seems to be about 11x11 in julia 0.5 with default optimizations.

Is probably much higher now.

David Sanders [6:47 PM]
Yes, my global optimization code in N dimensions now is super fast
(N-dimensional `IntervalBox`es just wrap an `SVector` of `Interval`s)
Something like a factor of 10 faster than with Julia 0.6
```

I think it might be worth re-evaluating the rule of thumb for the README since now the definition of "small" may have shifted. In another sense, "small" seems to have shifted high enough that now it's a discussion of runtime vs compilation time (compilation time can grow pretty fast for much larger static arrays, once again quoting @dpsanders 

>If I use `SVector`s of big dimension (e.g. 100, 200) there seems to be a huge compile-time penalty, but the calculations themselves are blindingly fast. Is this normal?

).  So a more nuanced discussion is probably needed in Julia v1.0 land, and I think this is important since that 11x11 number is something I and others copy around as a nice heuristic which may now be old!

**Comment by andyferris** [comment]
Yes... it's hard to condense down all the nuances into a sentence or two for new users. I'm definitely open to suggestions!

Out of curiousity, how high dimensional are these `IntervalBox`es? (I've been playing with my own interval boxes with a similar-sounding implementation, for implementation of R-trees for 2D and 3D geometries, but this seems orthogonal to large static arrays).

**Comment by andyferris** [comment]
OK, yep.

Also note #494. The long-term goal here has always been to use loops and so-on for larger arrays, which would solve the compilation time issue. For `N = 200` I would start to wonder if it makes sense to keep this data on the stack rather than the heap.

PS: random thought, what about the syntax `-10..10 ⊗ -10..10 ⊗ -10..10 ⊗ ...`?

**Comment by ChrisRackauckas** [comment]
MVector is still backed by a tuple? Its storage is still stack-allocated and it's just the wrapper type that gets allocated each time though?

>I also wonder about heap vs stack, but it doesn't seem to be something that's easy to play with?

DiffEq has both inplace and out-of-place routes, both optimized respectively. That would probably be a good "full program" perspective as to where it's a good spot to cut it off. We can make a test that is a PDE discretization of N points and find the switching point N.
